core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -332,6 +332,51 @@ package object config {
     ConfigBuilder("spark.dynamicAllocation.sustainedSchedulerBacklogTimeout")
       .fallbackConf(DYN_ALLOCATION_SCHEDULER_BACKLOG_TIMEOUT)
 
+  private[spark] val STREAMING_DYN_ALLOCATION_ENABLED =
```

Comment:
Could you add these to a new `Streaming.scala` instead? We shouldn't be adding more stuff to `package.scala`.

Reply:
@vanzin 

> If for some reason that is allowed, then the fix should take the max of both values.

This is not allowed to enable both of them. This check is done in the `scheduler/ExecutorAllocationManager`
https://github.com/apache/spark/blob/7beb464564a04004c638d6b0e1d8aafd9f038a0a/streaming/src/main/scala/org/apache/spark/streaming/scheduler/ExecutorAllocationManager.scala#L211

I agree to put these configs to `Streaming.scala`, it's more clear. I will update.